planner/core/find_best_task.go

```diff
@@ -2377,6 +2375,22 @@ func (ds *DataSource) convertToPointGet(prop *property.PhysicalProperty, candida
 		if partitionInfo == nil {
 			return invalidTask
 		}
+	} else if pi != nil {
+		// dynamic prune
+		idxs, err := PartitionPruning(ds.ctx, ds.table.GetPartitionedTable(), ds.allConds, ds.partitionNames, ds.TblCols, ds.names)
+		if err != nil {
+			return invalidTask
+		}
+		if len(idxs) == 1 && idxs[0] == FullRange {
+			if len(pi.Definitions) != 1 {
+				return invalidTask
+			}
+			partitionInfo = &pi.Definitions[0]
```

Comment:
I'm sure it already covered by https://github.com/pingcap/tidb/pull/45472/files#diff-4c4a5d9c358008fc0d3e898b87a14692668d4b60d949b2796d8b6ebc2b890720R236

Seems it has some problems.